pkg/k8s/watchers/pod.go

```diff
@@ -365,6 +365,14 @@ func (k *K8sWatcher) deletePodHostIP(pod *types.Pod) (bool, error) {
 		skipped bool
 	)
 
+	podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+	if ep := k.endpointManager.LookupPodName(podNSName); ep != nil {
+		// We don't care how many errors occurred, the callee will log them.
+		if errCount := k.endpointManager.DeleteEndpoint(ep); errCount > 0 {
+			return true, fmt.Errorf("encountered %d errors while deleting local pod", errCount)
+		}
```

Comment:
Some notes here:
1) the entire watcher will be blocked until this function is executed. I would suggest to do this in a go routine, is it possible?
2) In case of an error we are performing an early return and we won't delete the IPs from the ipache (basically the code from line 376 onward.

Reply:
For (2), if the endpoint is local then its corresponding ipcache entry is managed via the `sync-%d-identity-mapping` controller, see `runIPIdentitySync()`. From that I figured that we should either determine that the pod corresponds to a local endpoint and follow the endpoint deletion path, *or*, determine that the pod corresponds to a remote endpoint in which case we must do the ipcache update.

Though that said there's probably no harm in just spinning this out into a separate goroutine and allowing the ipcache delete attempt to occur.

Reply:
I fixed this up to run the logic in a separate goroutine, and added a new "cilium_endpoint_gc_count" metric so we can figure out whether any of these goroutines may be blocked for a long period.